src/include/txn.i

```diff
@@ -1088,18 +1103,28 @@ __wt_txn_update_check(WT_SESSION_IMPL *session, WT_UPDATE *upd)
     ignore_prepare_set = F_ISSET(txn, WT_TXN_IGNORE_PREPARE);
     F_CLR(txn, WT_TXN_IGNORE_PREPARE);
     for (; upd != NULL && !__wt_txn_upd_visible(session, upd); upd = upd->next) {
-        if (upd->txnid != WT_TXN_ABORTED) {
-            if (ignore_prepare_set)
-                F_SET(txn, WT_TXN_IGNORE_PREPARE);
-            WT_STAT_CONN_INCR(session, txn_update_conflict);
-            WT_STAT_DATA_INCR(session, txn_update_conflict);
-            return (__wt_txn_rollback_required(session, "conflict between concurrent operations"));
-        }
+        if (upd->txnid != WT_TXN_ABORTED)
+            WT_ERR(__txn_rollback_required(session));
```

Comment:
It would be better if you can structure the code here so that __txn_rollback_required is only done once in the function. It's a bit unusual to call a function that you know returns an error as a way to jump to the error handling, which makes this code hard for me to read - i.e: without knowing that __txn_rollback_required is different I look at this code and expect it to fall through and keep executing the rest of the function.

Reply:
Refactored using a local bool flag.